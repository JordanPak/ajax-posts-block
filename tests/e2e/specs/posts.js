/**
 * Test Posts block in editor
 *
 * @since .0.0
 */

import {
	activatePlugin,
	createNewPost,
	deactivatePlugin,
	enablePageDialogAccept,
	getEditedPostContent,
	insertBlock,
	publishPost,
	trashAllPosts,
} from '@wordpress/e2e-test-utils';

/**
 * Create a demo post and page
 */
const createDemoPosts = async () => {
	await createNewPost( { postType: 'post', title: 'APB Demo Post' } );
	await publishPost();

	await createNewPost( { postType: 'page', title: 'APB Demo Page' } );
	await publishPost();
};

/**
 * Trash posts and pages
 */
const trashPosts = async () => {
	await trashAllPosts();
	await trashAllPosts( 'page' );
};

describe( 'AJAX Posts Block', () => {
	// make sure editor dialogs are skipped and posts/pages are reset
	beforeAll( async () => {
		await activatePlugin( 'ajax-posts-block' );
		await enablePageDialogAccept();
		await trashPosts();
	} );

	afterAll( async () => {
		await trashPosts();
		await deactivatePlugin( 'ajax-posts-block' );
	} );

	it( 'Can be added', async () => {
		await createNewPost();
		await insertBlock( 'Posts' );

		// make sure it was inserted
		expect(
			await page.$( '[data-type="ajax-posts-block/posts"]' )
		).not.toBeNull();

		// and that it matches the snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Displays the demo post and page', async () => {
		await createDemoPosts();
		await createNewPost();
		await insertBlock( 'Posts' );

		const postTitles = await page.$$( '.apb-post-title' );
		const postTitlesText = [];

		for ( let i = 0; i < postTitles.length; i++ ) {
			postTitlesText.push(
				await (
					await postTitles[ i ].getProperty( 'innerText' )
				 ).jsonValue()
			);
		}

		expect( postTitlesText ).toStrictEqual( [
			'APB Demo Page',
			'APB Demo Post',
		] );
	}, 60000 ); // Demo post creation takes sooooooooo long :(
} );
