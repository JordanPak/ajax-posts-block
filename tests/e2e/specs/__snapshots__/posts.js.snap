// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AJAX Posts Block Can be added 1`] = `
"<!-- wp:ajax-posts-block/posts -->
<div data-num=\\"6\\" data-types=\\"\\" data-categories=\\"\\" data-tags=\\"\\" class=\\"wp-block-ajax-posts-block-posts\\"><p class=\\"apb-loading\\">Loading<svg aria-hidden=\\"true\\" data-apb-icon-spin=\\"true\\" data-apb-icon=\\"loading\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewbox=\\"0 0 512 512\\"><g fill=\\"currentColor\\"><path d=\\"M478.71 364.58zm-22 6.11l-27.83-15.9a15.92 15.92 0 01-6.94-19.2A184 184 0 11256 72c5.89 0 11.71.29 17.46.83-.74-.07-1.48-.15-2.23-.21-8.49-.69-15.23-7.31-15.23-15.83v-32a16 16 0 0115.34-16C266.24 8.46 261.18 8 256 8 119 8 8 119 8 256s111 248 248 248c98 0 182.42-56.95 222.71-139.42-4.13 7.86-14.23 10.55-22 6.11z\\" opacity=\\"0.4\\"></path><path d=\\"M271.23 72.62c-8.49-.69-15.23-7.31-15.23-15.83V24.73c0-9.11 7.67-16.78 16.77-16.17C401.92 17.18 504 124.67 504 256a246 246 0 01-25 108.24c-4 8.17-14.37 11-22.26 6.45l-27.84-15.9c-7.41-4.23-9.83-13.35-6.2-21.07A182.53 182.53 0 00440 256c0-96.49-74.27-175.63-168.77-183.38z\\"></path></g></svg></p></div>
<!-- /wp:ajax-posts-block/posts -->"
`;
